pkg/controllers/v1alpha1/dataprocess/implement.go

```diff
@@ -17,57 +17,213 @@
 package dataprocess
 
 import (
+	"context"
+	"fmt"
+	"time"
+
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
 	"github.com/fluid-cloudnative/fluid/pkg/dataoperation"
+	cdataprocess "github.com/fluid-cloudnative/fluid/pkg/dataprocess"
 	"github.com/fluid-cloudnative/fluid/pkg/runtime"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func (r *DataProcessReconciler) GetTargetDataset(object client.Object) (*datav1alpha1.Dataset, error) {
-	panic("not implemented") // TODO: Implement
+	dataProcess, ok := object.(*datav1alpha1.DataProcess)
+	if !ok {
+		return nil, fmt.Errorf("object %v is not of type DataProcess", object)
+	}
+
+	return utils.GetDataset(r.Client, dataProcess.Spec.Dataset.Name, dataProcess.Spec.Dataset.Namespace)
 }
 
 // GetReleaseNameSpacedName get the installed helm chart name
 func (r *DataProcessReconciler) GetReleaseNameSpacedName(object client.Object) types.NamespacedName {
-	panic("not implemented") // TODO: Implement
+	return types.NamespacedName{
+		Namespace: object.GetNamespace(),
+		Name:      utils.GetDataProcessReleaseName(object.GetName()),
+	}
 }
 
 // GetChartsDirectory get the helm charts directory of data operation
 func (r *DataProcessReconciler) GetChartsDirectory() string {
-	panic("not implemented") // TODO: Implement
+	return utils.GetChartsDirectory() + "/" + cdataprocess.DataProcessChart
 }
 
 // GetOperationType get the data operation type
 func (r *DataProcessReconciler) GetOperationType() dataoperation.OperationType {
-	panic("not implemented") // TODO: Implement
+	return dataoperation.DataProcess
 }
 
 // UpdateOperationApiStatus update the data operation status, object is the data operation crd instance.
 func (r *DataProcessReconciler) UpdateOperationApiStatus(object client.Object, opStatus *datav1alpha1.OperationStatus) error {
-	panic("not implemented") // TODO: Implement
+	dataProcess, ok := object.(*datav1alpha1.DataProcess)
+	if !ok {
+		return fmt.Errorf("%+v is not of type DataProcess", object)
+	}
+
+	var dataProcessCopy = dataProcess.DeepCopy()
+	dataProcessCopy.Status = *opStatus.DeepCopy()
+	return r.Status().Update(context.TODO(), dataProcessCopy)
 }
 
 // Validate check the data operation spec is valid or not, if not valid return error with conditions
 func (r *DataProcessReconciler) Validate(ctx runtime.ReconcileRequestContext, object client.Object) ([]datav1alpha1.Condition, error) {
-	panic("not implemented") // TODO: Implement
+	dataProcess, ok := object.(*datav1alpha1.DataProcess)
+	if !ok {
+		return []datav1alpha1.Condition{}, fmt.Errorf("object %v is not of type DataProcess", object)
+	}
+
+	// DataProcess's targetDataset must be in the same namespace.
+	if dataProcess.Namespace != dataProcess.Spec.Dataset.Namespace {
```

Comment:
How about defining processor interface?

```golang
type Processor interface{
     Validate(xxx)
    ValidateMount()
}
```

Reply:
Using `ProcessorImpl` interface instead. Thx for suggestion